contracts/Registry/TokenCuratedRegistry.sol

```diff
@@ -1,59 +1,103 @@
+  // there is not an active or passed challenge for this item. OwnedItemRegistry.remove
+  // requires that this is called by the item owner. LockableItemRegistry.remove requires
+  // that the item is not locked.
+  function remove(bytes32 id) public {
+    if (challengeExists(id)) {
+      require(challengeEnded(id) && !challengePassed(id));
+    }
+    delete challenges[id];
+    super.remove(id);
+  }
 
-  struct ItemCurationData {
-    uint applicationExpiry;
-    bool whitelisted;
-    address challengeAddress;
-    bool challengeResolved;
+  // Creates a new challenge for an item.
+  // Requires that the item exists, and that there is no existing challenge for the item.
+  // Requires msg.sender (the challenger) to match the owner's stake by transferring to
+  // this contract. The challenger's and owner's stake is transferred to the newly created
+  // challenge contract.
+  function challenge(bytes32 id) public {
+    require(!challengeExists(id));
+    require(token.transferFrom(msg.sender, this, minStake));
+    address challenge = challengeFactory.create(this, msg.sender, owners[id]);
+    require(token.transfer(challenge, minStake.mul(2)));
   }
 
-  constructor (
-    ERC20 _token,
-    uint _minStake,
-    uint _applicationPeriod,
-    ChallengeFactory _challengeFactory
-  ) public
-    StakedRegistry(_token, _minStake)
-  {
-    applicationPeriod = _applicationPeriod;
-    challengeFactory = _challengeFactory;
+  // Handles transfer of reward after a challenge has ended. Requires that there
+  // is an ended challenge for the item.
+  function resolveChallenge(bytes32 id) public {
+    require(challengeEnded(id));
+    uint reward = challenges[id].reward();
+    require(token.transferFrom(challenges[id], this, reward));
+    if (challengePassed(id)) {
+      // if the challenge passed, reward the challenger (via token.transfer) and remove
+      // the item.
+      require(token.transfer(challenges[id].challenger(), reward));
+      delete ownerStakes[id];
+      super.remove(id);
+    } else {
+      // if the challenge failed, reward the applicant (by adding to their staked balance)
+      ownerStakes[id] = ownerStakes[id].add(reward).sub(minStake);
+    }
+    delete unlockTimes[id];
+    delete challenges[id];
   }
 
-  function add(bytes32 data) public returns (bytes32) {
-    bytes32 id = keccak256(data);
-    itemsCurationData[id] = ItemCurationData(now + applicationPeriod, false, address(0), false);
-    return super.add(data);
+  // Returns true if the item exists and is not locked. We know that locked items are in
+  // the application phase, because the unlock time is set to now + applicationPeriod when
+  // items are added. Also, unlock time is set to 0 if an item is challenged and the
+  // challenge fails.
+  function inApplicationPhase(bytes32 id) public view returns (bool) {
+    require(exists(id));
+    return !isLocked(id);
```

Comment:
I would think this line would need to be `return isLocked(id);` (which would have it's own bugs). But right now this would return `true` for a whitelisted item, no?

Reply:
oops, good catch!

Reply:
ok even if you do correct this, this will return `true` for whitelisted items locked in a challenge

Reply:
oh or maybe not.... since a challenge doesn't lock necessarily....which is...interesting

Reply:
@emaG3m this is fixed